### ISSUE TITLE
[ Help ]

### ISSUE THREAD

**Issue by Hitlesama** [KaiserLancelot/kepub#5]
Hi KaiserLancelot, actually I don't have any knowledge about code but I really love read novel sfacg, can you help me how to download and use it to get text from sfacg plz :(((

**Comment by KaiserLancelot** [comment]
- First, make sure you are using a recent version of Debian/Ubuntu/WSL2 (other Linux systems have not been tested, but should work)
- The program requires AVX2 instruction set, the CPU is at least Intel Haswell (2013), if you use a virtual machine, you need to set additional CPU
- Then download the latest installation package [v2.2.0](https://github.com/KaiserLancelot/kepub/releases/download/v2.2.0/kepub-2.2.0-Linux.deb) and install
```bash
sudo dpkg -i kepub-2.2.0-Linux.deb
```
- Start downloading, for example
```
sfacg 400498
```

@Hitlesama

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/158358955-a2a6db0e-c8fb-445a-9ae1-c5b0732f80ea.png) I can't type password, I don't know why it happens ? :<

**Comment by KaiserLancelot** [comment]
At this time, you should enter the password of the computer user. By default, there is no echo. Please check whether the numeric keyboard is turned on.

@Hitlesama

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/158374770-0fe9bf75-c73a-4a17-a704-df35dcc98eb6.png)
![image](https://user-images.githubusercontent.com/101611167/158374846-4f34626e-ecac-4b72-a5ac-951435c48217.png)
Am I doing something wrong?

**Comment by KaiserLancelot** [comment]
> It may be that the working directory is incorrect.
>

**Comment by KaiserLancelot** [comment]
@Hitlesama

**Comment by Hitlesama** [comment]
So what should I do now

**Comment by Hitlesama** [comment]
Sorry for my limited knowledge of code

**Comment by KaiserLancelot** [comment]
Use 'ls' to show what files are in the current working directory, and use the 'cd' command to go to the specified directory. If you don't understand, please use the search engine to search for the basic use of Linux.

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/158419118-bbcfe650-7ec5-4054-a465-f4a4b4a6bce2.png)
... Help me plz ~~

**Comment by KaiserLancelot** [comment]
```bash
sudo apt install -y curl
curl -fsSL https://github.com/
```bash
If it doesn't work, the error is related to your computer/network; if it works, it may be a bug, but it may be difficult for me to reproduce and fix it. 

@Hitlesama

**Comment by KaiserLancelot** [comment]
If you really can't solve it and you want, you can send your username, password and book-id to my email, and I will download it for you in my free time. 

@Hitlesama

**Comment by KaiserLancelot** [comment]
> Finally, I downloaded my favorite book completely :> Thanks for your help

Enjoy yourself

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/158784187-aad42b70-a5ed-4218-a2b6-cb71bbbeb225.png)
Could you please insert characters before the chapter title like the picture above?

**Comment by KaiserLancelot** [comment]
By default, '[WEB] ' will be inserted before the title. If this does not meet your needs, just replace it with a text editor (such as VSCode).

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/159103701-ee597677-21c5-416d-96c7-8f28030ca777.png)
I updated to ver 2.4.0, then download books and I got this error :(((

**Comment by KaiserLancelot** [comment]
Please take a look at the previous output, if there is a warning about not being able to download this image
@Hitlesama

**Comment by KaiserLancelot** [comment]
Your system seems to lack fonts and cannot display Chinese well, you can try [source-han-serif](https://github.com/adobe-fonts/source-han-serif) and [source-han-sans ](https://github.com/adobe-fonts/source-han-sans)

**Comment by Hitlesama** [comment]
Hmm...I think ver 2.4.0 has problem like picture above because I just try downgrade kepub from 2.4.0 to 2.3.0 and download this book. As a result, it downloaded completely without error. The error of not being able to download this image made me can't get text too~~

**Comment by KaiserLancelot** [comment]
I think [v2.4.1](https://github.com/KaiserLancelot/kepub/releases/download/v2.4.1/kepub-2.4.1-Linux.deb) has fixed this, try it out and let me know if it works 
@Hitlesama

**Comment by Hitlesama** [comment]
@KaiserLancelot Okie, I just try in v2.4.2 and it downloaded text completely but it still has some problems about image like this 
![image](https://user-images.githubusercontent.com/101611167/159106630-715cdff5-7e2c-4a6e-94de-0a5571e32a6e.png)
In v2.3.0 there's no warning about Image not exists.

**Comment by KaiserLancelot** [comment]
What is the book-id of this book? I will try to reproduce it
@Hitlesama

**Comment by Hitlesama** [comment]
@KaiserLancelot sfacg 400179

**Comment by KaiserLancelot** [comment]
> @KaiserLancelot sfacg 400179

It has been confirmed that the case where the same image is referenced in different chapters is not handled.

[v2.4.3](https://github.com/KaiserLancelot/kepub/releases/download/v2.4.3/kepub-2.4.3-Linux.deb) fixed 

@Hitlesama

**Comment by Hitlesama** [comment]
@KaiserLancelot Okay, I wonder how to download ciweimao book

**Comment by KaiserLancelot** [comment]
Same as sfacg
@Hitlesama

**Comment by Hitlesama** [comment]
Thanks, more over, I suggest that the download speed of ciweimao must be improved, When I download one of ciweimao,  I feel that ciweimao slower than sfacg lots of, in my opinion. Sorry, is it pissed you off for my suggest :<

**Comment by KaiserLancelot** [comment]
If you feel slow, use multi-threaded download 
```bash
ciweimao 100012892 -m4
```
Be careful, don't set the maximum concurrency too high, and don't download too many times in a short period of time 

@Hitlesama

**Comment by Hitlesama** [comment]
Anou, I don't know how to set the maximum concurrency  and what's " -m4" . Could you rephrase clearly plz ?

**Comment by KaiserLancelot** [comment]
You only need to know that adding the option '-m4' can speed up the download speed. The number after 'm' is the maximum concurrent number. The larger the number, the faster the download. However, if the download is too fast, it may lead to a ban, please use it with caution.

**Comment by Hitlesama** [comment]
@KaiserLancelot Oh, so "-m4" is the best option and safe for account nah

**Comment by KaiserLancelot** [comment]
Not sure, it depends on your network condition, the average time to download a chapter should be no less than 50ms
@Hitlesama

**Comment by Hitlesama** [comment]
I got it, and I hope that you can try study how to download text from b.faloo in the future :D I also love some books in that website

**Comment by Hitlesama** [comment]
![image](https://user-images.githubusercontent.com/101611167/160265620-3a6146bf-eee8-4e89-b007-4290a646027c.png)
The above-mentioned issue exists in version 2.4.5. 
@KaiserLancelot

**Comment by KaiserLancelot** [comment]
I will fix it in the next version. Please use the previous version first. Also, please open a new issue for new problems.
@Hitlesama

**Comment by Hitlesama** [comment]
Okay
